content/docs/for-platform-operators/tutorials/lifecycle/_index.md

```diff
@@ -5,4 +5,434 @@ expand=true
+
+    Let’s do that for every buildpack in the `samples/buildpacks` directory:
+
+    ```text
+    for f in $(ls --color=no ${CNB_SAMPLES_PATH}/buildpacks | grep -v README)
+    do
+    bp_version=$(cat ${CNB_SAMPLES_PATH}/buildpacks/$f/buildpack.toml | dasel -r toml buildpack.version | sed s/\'//g);
+    mkdir -p ./buildpacks/samples_"${f}"/${bp_version}
+    cp -r "$CNB_SAMPLES_PATH/buildpacks/${f}/*" ./buildpacks/samples_"${f}"/${bp_version}/
+    done
+    ```
+
+Now, you can run the `detector` binary:
+
+```text
+${CNB_LIFECYCLE_PATH}/detector -log-level debug -layers="./layers" -order="./order.toml" -buildpacks="./buildpacks" -app apps/bash-script
+```
+
+The output of the above command should have `layers/group.toml` and `layers/plan.toml` output files (i.e., the groups that have passed the detection have been written into the `group.toml` file writing its build plan into the `plan.toml` file)
+
+```text
+Starting detector...
+Parsing inputs...
+Ensuring privileges...
+Executing command...
+Timer: Detector started at 2024-10-01T07:00:50Z
+Checking for match against descriptor: {linux   []}
+target distro name/version labels not found, reading /etc/os-release file
+Checking for match against descriptor: {linux   []}
+target distro name/version labels not found, reading /etc/os-release file
+Checking for match against descriptor: {linux   []}
+target distro name/version labels not found, reading /etc/os-release file
+Checking for match against descriptor: {linux   []}
+target distro name/version labels not found, reading /etc/os-release file
+Checking for match against descriptor: {linux   []}
+======== Results ========
+fail: samples/java-maven@0.0.2
+======== Results ========
+fail: samples/kotlin-gradle@0.0.2
+======== Results ========
+fail: samples/ruby-bundler@0.0.1
+======== Results ========
+pass: samples/hello-world@0.0.1
+pass: samples/hello-moon@0.0.1
+Resolving plan... (try #1)
+samples/hello-world 0.0.1
+samples/hello-moon  0.0.1
+Timer: Detector ran for 26.011769ms and ended at 2024-10-01T07:00:50Z
+```
+
+You can view more details about the [order](https://buildpacks.io/docs/for-platform-operators/concepts/lifecycle/detect/#ordertoml), [group](https://buildpacks.io/docs/for-platform-operators/concepts/lifecycle/detect/#grouptoml) and [plan](https://buildpacks.io/docs/concepts/components/lifecycle/detect/#plantoml) toml files in the platform documentation.
+
+#### Restore
+
+The `restorer` retrieves cache contents, if it contains any, into the build container. During this phase, the `restorer` looks for layers that could be reused or should be replaced while building the application image.
+
+First, you need to create the `cache` directory, and then run the `restorer` binary as added below:
+
+```text
+mkdir cache
+```
+
+```text
+${CNB_LIFECYCLE_PATH}/restorer -log-level debug -layers="./layers" -group="./layers/group.toml" -cache-dir="./cache" -analyzed="./layers/analyzed.toml"
```

Comment:
```suggestion
${CNB_LIFECYCLE_PATH}/restorer -log-level debug -daemon -layers="./layers" -group="./layers/group.toml" -cache-dir="./cache" -analyzed="./layers/analyzed.toml"
```

Reply:
This fixes `ERROR: failed to resolve inputs: run image not found in analyzed metadata`